pkg/sdk/testint/connections_gen_integration_test.go

```diff
@@ -0,0 +1,352 @@
+package testint
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/stretchr/testify/require"
+
+	assertions "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert/objectassert"
+)
+
+func TestInt_Connections(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	orgName, err := client.ContextFunctions.CurrentOrganizationName(ctx)
+	require.NoError(t, err)
+
+	accountName, err := client.ContextFunctions.CurrentAccountName(ctx)
+	require.NoError(t, err)
+
+	region, err := client.ContextFunctions.CurrentRegion(ctx)
+	require.NoError(t, err)
+
+	t.Run("Create minimal", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		err := client.Connections.Create(ctx, sdk.NewCreateConnectionRequest(id))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Connection.DropFunc(t, id))
+
+		assertions.AssertThatObject(t, objectassert.Connection(t, id).
+			HasSnowflakeRegion(region).
+			HasAccountName(accountName).
+			HasName(id.Name()).
+			HasNoComment().
+			HasIsPrimary(true).
+			HasPrimary(fmt.Sprintf("%s.%s.%s", orgName, accountName, id.Name())).
+			HasFailoverAllowedToAccounts(
+				[]string{
+					fmt.Sprintf("%s.%s", orgName, accountName),
+				},
+			).
+			HasOrganizationName(orgName).
+			HasAccountLocator(client.GetAccountLocator()),
+		)
+	})
+
+	t.Run("Create all options", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		err := client.Connections.Create(ctx, sdk.NewCreateConnectionRequest(id).
+			WithIfNotExists(true).
+			WithComment("test comment for connection"))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Connection.DropFunc(t, id))
+
+		assertions.AssertThatObject(t, objectassert.Connection(t, id).
+			HasSnowflakeRegion(region).
+			HasAccountName(accountName).
+			HasName(id.Name()).
+			HasComment("test comment for connection").
+			HasIsPrimary(true).
+			HasPrimary(fmt.Sprintf("%s.%s.%s", orgName, accountName, id.Name())).
+			HasFailoverAllowedToAccounts(
+				[]string{
+					fmt.Sprintf("%s.%s", orgName, accountName),
+				},
+			).
+			HasOrganizationName(orgName).
```

Comment:
What about connection_url? I guess we can just check if it's non-empty.

Reply:
we can check it, with fmt.Sprintf() but sitll it can be checked